cognite/neat/graph/stores/_base.py

```diff
@@ -179,17 +179,27 @@ def read(self, class_: str) -> Iterable[tuple[str, dict[str, list[str]]]]:
             warnings.warn("Desired type not found in graph!", stacklevel=2)
             return None
 
-        if InformationAnalysis(self.rules).has_hop_transformations():
+        if (has_hop_transformations := InformationAnalysis(self.rules).has_hop_transformations()) or (
+            has_all_reference_transformations := InformationAnalysis(self.rules).has_all_reference_transformations()
+        ):
+            msg = (
+                f"Rules contain [{'Hop' if has_hop_transformations else '' }"
+                f", {'AllReference' if has_all_reference_transformations else '' }]"
+                " rdfpath."
+                f" Run [{'ReduceHopTraversal' if has_hop_transformations else '' }"
+                f", {'AddAllReferences' if has_all_reference_transformations else '' }]"
+                " transformer(s) first!"
+            )
```

Comment:
If the rules has_hop_transformation, then the message will raise an Exception as `has_all_reference_transformation` will never have been created. Suggest doing these two calculations ahead of the check.
```suggestion
        analysis = InformationAnalysis(self.rules)
        has_hop_transformations = analysis.has_hop_transformations()
        has_all_reference_transformations = analysis.has_all_reference_transformations()
        if has_hop_transformations or has_all_reference_transformations:
            msg = (
                f"Rules contain [{'Hop' if has_hop_transformations else '' }"
                f", {'AllReference' if has_all_reference_transformations else '' }]"
                " rdfpath."
                f" Run [{'ReduceHopTraversal' if has_hop_transformations else '' }"
                f", {'AddAllReferences' if has_all_reference_transformations else '' }]"
                " transformer(s) first!"
            )
```